.github/workflows/prepare-release.yml

```diff
@@ -19,8 +19,35 @@ on:
         required: true
         description: Current version (beta, like 0.69.1). Don't include `v`.
 jobs:
+  #validate-version format
+  validate-versions:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Validate version format
+      run: |
+        validate_version() {
+          local regex_pattern_beta='^[0-9]+\.[0-9]+\.[0-9]+$'
+          local regex_pattern_stable='^[1-9][0-9]*\.[0-9]+\.[0-9]+$'
+
+          if [[ ! "$1" =~ $regex_pattern_beta ]]; then
+            if [[ ! "$1" =~ $regex_pattern_stable ]]; then
+              echo "Invalid $2 version format. For beta, it can be 0.1.0 or higher. For stable, Major version must be greater than 1."
+              exit 1
+            fi
+          fi
```

Comment:
Instead of nested if conditions, you could create 2 functions, one called `validate_beta_version` and 1 called `validate_stable_version` that handle their specific regexes and error message.  Then at the bottom of the script call the specific functions.